### PR TITLE
feat: friendlier 'make'

### PR DESCRIPTION
**Pull request by leseb**
Now 'make' checks for commands being present or not before running targets. It handles missing commands nicely.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -123,34 +126,47 @@ rocm-gfx900: $(ROCM_DEPS)  ## Build container for AMD ROCm gfx900 (untested)
...
 .PHONY: man
-man:
+man: check-tox
 	tox -e docs
 
 .PHONY: docs
 docs: man ## Run tox -e docs against code
 
+##@ Validation
+check-tox:
```
Comment (tiran) [inline Makefile:158]: The check targets should be marked as phony targets. They do not create files with the target name.

File: Makefile [hunk 2]
```diff
@@ -64,7 +67,7 @@ cuda: $(CUDA_DEPS)  ## Build container for NVidia CUDA
 # The base container uses uids beyond 65535. Rootless builds may not work
 # unless the user account has extended subordinate ids up to 2**24 - 1.
 .PHONY: hpu
-hpu: $(HPU_DEPS)  ## Build container for Intel Gaudi HPU
+hpu: check-engine $(HPU_DEPS)  ## Build container for Intel Gaudi HPU
```
Comment (tiran) [inline Makefile:70]: Please move `check-engine` to the end of the dependency list. The first item `$<` has to be the container file.

**Review by tiran — APPROVED** [review]
LGTM

You have to rebase after #1162 has landed.

**Comment by nathan-weinberg** [comment]
@Mergifyio rebase

**Review by russellb — APPROVED** [review]

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**